frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx

```diff
@@ -0,0 +1,535 @@
+        },
+      });
+      expect(screen.getByText("b@metabase.test")).toBeInTheDocument();
+
+      userEvent.type(screen.getByLabelText("Filter value"), "a");
+      act(() => jest.advanceTimersByTime(1000));
+      userEvent.click(await screen.findByText("a@metabase.test"));
+
+      expect(onChange).toHaveBeenLastCalledWith([
+        "b@metabase.test",
+        "a@metabase.test",
+      ]);
+    });
+
+    it("should handle field values remapping", async () => {
+      const { onChange } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+        searchValues: {
+          a: createMockFieldValues({
+            field_id: PEOPLE.EMAIL,
+            values: [["a-test", "a@metabase.test"]],
+          }),
+        },
+      });
+
+      userEvent.type(screen.getByPlaceholderText("Search by Email"), "a");
+      act(() => jest.advanceTimersByTime(1000));
+      userEvent.click(await screen.findByText("a@metabase.test"));
+
+      expect(onChange).toHaveBeenLastCalledWith(["a-test"]);
+    });
+  });
+
+  describe("no values", () => {
+    const column = findColumn("PEOPLE", "PASSWORD");
+
+    it("should allow to add a value", async () => {
+      const { onChange, onFocus, onBlur } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+      });
+
+      userEvent.type(screen.getByPlaceholderText("Enter some text"), "abc");
+      userEvent.tab();
+
+      expect(onFocus).toHaveBeenCalled();
+      expect(onChange).toHaveBeenLastCalledWith(["abc"]);
+      expect(onBlur).toHaveBeenCalled();
+    });
+
+    it("should allow to add multiple values", async () => {
+      const { onChange, onFocus, onBlur } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: ["abc"],
+      });
+
+      userEvent.type(screen.getByLabelText("Filter value"), "bce");
+      userEvent.tab();
+
+      expect(onFocus).toHaveBeenCalled();
+      expect(onChange).toHaveBeenLastCalledWith(["abc", "bce"]);
+      expect(onBlur).toHaveBeenCalled();
+    });
+
+    it("should not allow to add empty values", async () => {
+      const { onChange, onFocus, onBlur } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+      });
+
+      const input = screen.getByPlaceholderText("Enter some text");
+      userEvent.type(input, "abc");
+      userEvent.clear(input);
+      userEvent.tab();
+
+      expect(onFocus).toHaveBeenCalled();
+      expect(onChange).toHaveBeenLastCalledWith([]);
+      expect(onBlur).toHaveBeenCalled();
+    });
+
+    it("should allow to remove a value", async () => {
+      const { onChange } = await setupStringPicker({
+        query,
+        stageIndex,
+        column,
+        values: ["abc", "bce"],
+      });
+
+      userEvent.type(screen.getByLabelText("Filter value"), "{backspace}");
+
+      expect(onChange).toHaveBeenLastCalledWith(["abc"]);
```

Comment:
I think it's worth asserting if it's possible to clear the value:

```suggestion
      expect(onChange).toHaveBeenLastCalledWith(["abc"]);
      
      userEvent.type(screen.getByLabelText("Filter value"), "{backspace}");

      expect(onChange).toHaveBeenLastCalledWith([]);
```

Reply:
Fixed